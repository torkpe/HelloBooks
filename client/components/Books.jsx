import React, { Component } from 'react';
import { Link } from 'react-router';

/**
 *
 * @param {object} props
 * @return {XML} JSX
 */
const Books = (props) => {
  const { book } = props;
  const mdlStyle = `
    mdl-button
    mdl-button--colored
    mdl-js-button
    mdl-js-ripple-effect
  `;
  return (
    <div className="mdl-cell mdl-cell--4-col">
      <div className="demo-card-square mdl-card mdl-shadow--2dp home-card contents">
        <div className="mdl-card__title mdl-card--expand">
          <h2 className="mdl-card__title-text">{book.title}</h2>
        </div>
        <img src={book.cover} alt="Book cover" />
        <div className="mdl-card__supporting-text">
          {book.description}
        </div>
        <div className="mdl-card__actions mdl-card--border">
          <Link to={`single/${book.id}`} className={mdlStyle}>
              Details
          </Link>
        </div>
      </div>
    </div>
  );
};
export default Books;
