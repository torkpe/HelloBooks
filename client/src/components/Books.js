import React, { Component } from 'react';

class Books extends Component {
  render() {
    const book = this.props.book
    const userId = this.props.userId
    const data={
      user: this.props.name,
      userId,
      type: 'admin',
      message: `${this.props.name} just borrowed the book <a href='https://hellobooks-project.herokuapp.com/api/books/${book.id}'>${book.title}</a>`
    }
    const data2={
      user: this.props.name,
      userId,
      type: 'adnin',
      message: `${this.props.name} just returned the book <a href='https://hellobooks-project.herokuapp.com/api/books/${book.id}'>${book.title}</a>`
    }
    const borrow=(e)=>{
      e.preventDefault()
      this.props.borrowBook(userId, book.id, data)
    }
    const returnBorrowed=(e)=>{      
      this.props.returnBook(userId, book.id, data2)
    }
    const chargeUser = (e) => {
      this.props.charge(4, 5)
    }
    if(this.props.successfullyBorrowed){
      this.props.notify(data)
    }
    return (
      <div className="mdl-cell mdl-cell--4-col">
        <div className="demo-card-square mdl-card mdl-shadow--2dp home-card">
          <div className="mdl-card__title mdl-card--expand">
            <h2 className="mdl-card__title-text">{book.title}</h2>
          </div>
            <img src={book.cover} alt="Book cover"/>
          <div className="mdl-card__supporting-text">
            {book.description}
          </div>
          <div className="mdl-card__actions mdl-card--border">
            { this.props.charge ? '' :
            <a onClick={borrow}
            className="mdl-button mdl-button--colored mdl-js-button mdl-js-ripple-effect">
            borrow book
            </a> }
            { this.props.charge ? '' :
            <a onClick={returnBorrowed}
            className="mdl-button mdl-button--colored mdl-js-button mdl-js-ripple-effect">              
              return book
            </a>}
            { this.props.charge ?
              <a onClick={chargeUser} className="mdl-button mdl-button--colored mdl-js-button mdl-js-ripple-effect">              
                chage
            </a>
            : ''}
          </div>
        </div>
      </div>
    );
  }
}
export default Books;
