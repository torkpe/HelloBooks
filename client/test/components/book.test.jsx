import React from 'react';
import toJson from 'enzyme-to-json';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import render from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import Books from '../../components/Books.jsx';
import storeConfiguration from '../../store/store';

configure({ adapter: new Adapter() });

describe('<Home />', () => {
  const props = {
    getBooks: jest.fn(),
    book: {
      title: 'eze goes to school',
      description: 'a boy from the village who finally went to school',
      id: 1,
      cover: 'xcgvhjkhcvbkjhgcvbjhcv',
      pdf: 'cghjklhgcfhgjkhgjh',
      quantity: 1,
      genre: 'eductaional'
    },
  };
  it('renders <Home /> component', () => {
    const shallowComponent = shallow(<Books {...props} />);
    const tree = toJson(shallowComponent);
    expect(tree).toMatchSnapshot();
  });
});
