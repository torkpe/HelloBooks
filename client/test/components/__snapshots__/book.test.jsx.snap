// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders <Home /> component 1`] = `
<div
  className="mdl-cell mdl-cell--4-col"
>
  <div
    className="demo-card-square mdl-card mdl-shadow--2dp home-card contents"
  >
    <div
      className="mdl-card__title mdl-card--expand"
    >
      <h2
        className="mdl-card__title-text"
      >
        eze goes to school
      </h2>
    </div>
    <img
      alt="Book cover"
      src="xcgvhjkhcvbkjhgcvbjhcv"
    />
    <div
      className="mdl-card__supporting-text"
    >
      a boy from the village who finally went to school
    </div>
    <div
      className="mdl-card__actions mdl-card--border"
    >
      <Link
        className="
    mdl-button
    mdl-button--colored
    mdl-js-button
    mdl-js-ripple-effect
  "
        onlyActiveOnIndex={false}
        style={Object {}}
        to="single/1"
      >
        Details
      </Link>
    </div>
  </div>
</div>
`;
